osbs/build/build_requestv2.py

```diff
@@ -0,0 +1,395 @@
+        self.scratch = kwargs.pop('scratch', False)
+        # When true, it indicates build was automatically started by
+        # OpenShift via a trigger, for instance ImageChangeTrigger
+        self.is_auto = kwargs.pop('is_auto', False)
+        # An isolated build is meant to patch a certain release and not
+        # update transient tags in container registry
+        self.isolated = kwargs.pop('isolated', False)
+
+        self.validate_build_variation()
+
+        self.base_image = kwargs.get('base_image')
+        self.platform_node_selector = kwargs.get('platform_node_selector', {})
+        self.scratch_build_node_selector = kwargs.get('scratch_build_node_selector', {})
+        self.explicit_build_node_selector = kwargs.get('explicit_build_node_selector', {})
+        self.auto_build_node_selector = kwargs.get('auto_build_node_selector', {})
+        self.isolated_build_node_selector = kwargs.get('isolated_build_node_selector', {})
+
+        logger.debug("setting params '%s' for %s", kwargs, self.user_params)
+        self.user_params.set_params(**kwargs)
+
+    def validate_build_variation(self):
+        variations = (self.scratch, self.is_auto, self.isolated)
+        if variations.count(True) > 1:
+            raise OsbsValidationException(
+                'Build variations are mutually exclusive. '
+                'Must set either scratch, is_auto, isolated, or none. ')
+
+    def set_resource_limits(self, cpu=None, memory=None, storage=None):
+        if self._resource_limits is None:
+            self._resource_limits = {}
+
+        if cpu is not None:
+            self._resource_limits['cpu'] = cpu
+
+        if memory is not None:
+            self._resource_limits['memory'] = memory
+
+        if storage is not None:
+            self._resource_limits['storage'] = storage
+
+    def set_repo_info(self, repo_info):
+        self._repo_info = repo_info
+
+    @property
+    def build_id(self):
+        return self.build_json['metadata']['name']
+
+    @property
+    def template(self):
+        if self._template is None:
+            path = os.path.join(self.build_json_store, self._outer_template_path)
+            logger.debug("loading template from path %s", path)
+            try:
+                with open(path, "r") as fp:
+                    self._template = json.load(fp)
+            except (IOError, OSError) as ex:
+                raise OsbsException("Can't open template '%s': %s" %
+                                    (path, repr(ex)))
+        return self._template
+
+    # Override
+    @property
+    def inner_template(self):
+        raise RuntimeError('inner_template not supported in BuildRequestV2')
+
+    # Override
+    @property
+    def customize_conf(self):
+        raise RuntimeError('customize_conf not supported in BuildRequestV2')
+
+    # Override
+    @property
+    def dj(self):
+        raise RuntimeError('DockJson not supported in BuildRequestV2')
+
+    def has_ist_trigger(self):
```

Comment:
No need to override this method.

I won't comment on others, but if the method is the same as in parent class we don't need to override it.